concordium-consensus/tests/scheduler/SchedulerTests/BakerTransactions.hs

```diff
@@ -1,66 +1,56 @@
 {-# LANGUAGE DataKinds #-}
 {-# LANGUAGE NumericUnderscores #-}
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
 
 module SchedulerTests.BakerTransactions where
 
+import Control.Monad
+import Data.Bifunctor (first)
+import qualified Data.List as List
+import Lens.Micro.Platform
+import System.Random
 import Test.Hspec
 
 import Concordium.GlobalState.BakerInfo
-import Concordium.GlobalState.Basic.BlockState
-import Concordium.GlobalState.Basic.BlockState.Accounts as Acc
-import Concordium.GlobalState.Basic.BlockState.Bakers
-import Concordium.GlobalState.Basic.BlockState.Invariants
+import qualified Concordium.GlobalState.Basic.BlockState.Account as Basic
+import qualified Concordium.GlobalState.BlockState as BS
+import qualified Concordium.GlobalState.Persistent.BlockState as BS
 import qualified Concordium.Scheduler as Sch
-import qualified Concordium.Scheduler.EnvironmentImplementation as Types
 import Concordium.Scheduler.Runner
 import qualified Concordium.Scheduler.Types as Types
-import Concordium.TransactionVerification
 import Concordium.Types.Accounts (
     bakerAggregationVerifyKey,
     bakerElectionVerifyKey,
     bakerInfo,
     bakerSignatureVerifyKey,
  )
-import Control.Monad
-import Data.Foldable
-import qualified Data.Map.Strict as Map
-import Data.Maybe
-import System.Random
 
 import qualified Concordium.Crypto.BlockSignature as BlockSig
 import qualified Concordium.Crypto.BlsSignature as Bls
+import Concordium.Crypto.DummyData
 import qualified Concordium.Crypto.SignatureScheme as SigScheme
 import qualified Concordium.Crypto.VRF as VRF
-
-import Lens.Micro.Platform
-
-import Concordium.Crypto.DummyData
 import Concordium.GlobalState.DummyData
 import Concordium.Scheduler.DummyData
 import Concordium.Types.DummyData
+import System.IO.Unsafe
 
-import SchedulerTests.Helpers
+import qualified SchedulerTests.Helpers as Helpers
 import SchedulerTests.TestUtils
 
-import qualified Concordium.GlobalState.Basic.BlockState.LFMBTree as L
-
-shouldReturnP :: Show a => IO a -> (a -> Bool) -> IO ()
-shouldReturnP action f = action >>= (`shouldSatisfy` f)
-
 keyPair :: Int -> SigScheme.KeyPair
 keyPair = uncurry SigScheme.KeyPairEd25519 . fst . randomEd25519KeyPair . mkStdGen
 
 account :: Int -> Types.AccountAddress
 account = accountAddressFrom
 
-initialBlockState :: BlockState PV1
-initialBlockState =
-    createBlockState $
-        foldr
-            putAccountWithRegIds
-            Acc.emptyAccounts
-            [mkAccount (SigScheme.correspondingVerifyKey (keyPair i)) (account i) 400_000_000_000 | i <- reverse [0 .. 3]]
+accounts :: (Types.IsAccountVersion av) => [Basic.Account av]
```

Comment:
why are we still having basic accounts? Should we not have `mkAccount` helper that constructs persistent accounts instead?

Reply:
ok.